src/Caliburn.Micro.Platform/xforms/NavigationPageAdapter.cs

```diff
@@ -19,6 +19,26 @@ public class NavigationPageAdapter : INavigationService {
             this.navigationPage = navigationPage;
         }
 
+
+        /// <summary>
+        /// Allow Xamarin to navigate to a ViewModel backed by a view which is of type <see cref="T:Xamarin.Forms.ContentView"/> by adapting the result
+        /// to a <see cref="T:Xamarin.Forms.ContentPage"/>.
+        /// </summary>
+        /// <param name="view">The view to be adapted</param>
+        /// <param name="viewModel">The view model which is bound to the view</param>
+        /// <returns>The adapted ContentPage</returns>
+        protected virtual ContentPage CreateContentPage(ContentView view, object viewModel)
+        {
+            var page = new ContentPage { Content = view };
+
+            var hasDiplayName = viewModel as IHaveDisplayName;
+            if (hasDiplayName != null) {
+                page.Title = hasDiplayName.DisplayName;
```

Comment:
Not sure if it's applicable to Xamarin Forms, but should `Title` be binded to `DisplayName` to reflect any future changes of the property?